### PR TITLE
Remove the SIP version of PyKDL

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
I don't see any reason to keep supporting the SIP version of PyKDL. As SIP doesn't provide any benefits on top of PyBind, but maintaining both is extra work. PyBind provides some features, which are not included in SIP. Therefore I think PyBind is the way to go.

I.E. I wanted to add the extended constructor of a solver, requested in https://github.com/orocos/orocos_kinematics_dynamics/issues/292. In PyBind this is a very simple solution, but SIP doesn't support EIGEN classes out of the box and a custom constructor needs to be created.

### CODE REVIEW

File: python_orocos_kdl/INSTALL.md
Comment (meyerj) [inline python_orocos_kdl/INSTALL.md:42]:
```suggestion
    - Python 2: `python2 ../tests/PyKDLtest.py`
    - Python 3: `python3 ../tests/PyKDLtest.py`
```

**Review by meyerj — COMMENTED** [review]
I agree that maintaining both is likely not necessary and PyBind works well on all systems I tested newer versions with. Also the documentation and installation instructions update are pretty useful. Thanks for your effort!

Are there any known disadvantages of PyBind, i.e.
- Does it require C++11 or higher or newer versions of any dependency than with SIP? If yes, removing SIP basically implies that we drop support for older platforms, at least for `python_orocos_kdl`.
- Has it been tested on macOS or Windows?

I have not reviewed this patch in detail, but it looks straight-forward. The installation instruction are tailored towards Debian/Ubuntu. That could be mentioned, without the need to document how to install dependencies on other platforms in detail.

**Review by smits — APPROVED** [review]
Looks good to me.

**Comment by MatthijsBurgh** [comment]
This has not been tested on MacOS and/or Windows. But pybind11 does support MacOS and Windows, https://pybind11.readthedocs.io/en/stable/basics.html?highlight=windows.

Have we ever supported anything outside Linux? If we support other OS, we need to have CI for it.

According to GCC, from 4.8 it support c++11. GCC 4.8 is available in 14.04. So That is the oldest LTS that is running, so I think it is fair to drop support for older OS.

**Comment by MatthijsBurgh** [comment]
@meyerj I will wait for your approval before merging

**Review by meyerj — APPROVED** [review]

**Pull request merged**
